src/main/java/io/xstefank/wildfly/bot/util/GithubProcessor.java

```diff
@@ -0,0 +1,250 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Random;
+import java.util.Set;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+public class GithubProcessor {
+
+    public static final String COLLABORATOR_MISSING_SUBJECT = "Missing collaborator in the %s repository";
+
+    public static final String COLLABORATOR_MISSING_BODY = """
+        Hello,
+
+        The following people are not collaborators in the repository %s, however they were requested to review Pull Request number #%d
+
+        %s
+
+        ---
+        This is generated message, please do not respond.""";
+
+    private static final Logger LOG = Logger.getLogger(GithubProcessor.class);
+    private Pattern SKIP_FORMAT_COMMAND;
+
+    @Inject
+    WildFlyBotConfig wildFlyBotConfig;
+
+    @Inject
+    Mailer mailer;
+
+    @ConfigProperty(name = "quarkus.mailer.username")
+    Optional<String> username;
+
+    @PostConstruct
+    void construct() {
+        SKIP_FORMAT_COMMAND = Pattern.compile("@%s skip format".formatted(wildFlyBotConfig.githubName()), Pattern.DOTALL | Pattern.LITERAL);
+    }
+
+    public void commitStatusSuccess(GHPullRequest pullRequest, String checkName, String description) throws IOException {
+        String sha = pullRequest.getHead().getSha();
+
+        if (wildFlyBotConfig.isDryRun()) {
+            LOG.infof("Pull request #%d - Commit status success {%s, %s, %s}", pullRequest.getNumber(), sha, checkName, description);
+        } else {
+            pullRequest.getRepository().createCommitStatus(sha, GHCommitState.SUCCESS, "", description, checkName);
+        }
+    }
+
+    public void commitStatusError(GHPullRequest pullRequest, String checkName, String description) throws IOException {
+        String sha = pullRequest.getHead().getSha();
+
+        if (wildFlyBotConfig.isDryRun()) {
+            LOG.infof("Pull request #%d - Commit status failure {%s, %s, %s}", pullRequest.getNumber(), sha, checkName, description);
+        } else {
+            pullRequest.getRepository().createCommitStatus(sha, GHCommitState.ERROR, "", description, checkName);
+        }
+    }
+
+    public void processMentions(GHPullRequest pullRequest, GitHub gitHub, Map<String, Boolean> mentions, List<String> emails) throws IOException {
+        if (mentions.isEmpty()) {
+            updateCCMentions(pullRequest, Collections.emptyList());
+            return;
+        }
+
+        Set<String> collaborators = pullRequest.getRepository().getCollaboratorNames();
+        List<String> ccMentions = mentions.entrySet()
+                .stream()
+                .filter(o -> !o.getValue())
+                .map(Map.Entry::getKey)
+                .collect(Collectors.toList());
+
+        List<String> reviewers = mentions.entrySet()
+                .stream()
+                .filter(Map.Entry::getValue)
+                .map(Map.Entry::getKey)
+                .collect(Collectors.toList());
```

Comment:
Sure, I can update it so. However it might result in lower performance, as we wanted to keep the order of ccMentions, thus using a List. However to not add duplicates we would need to check this with `contains()` with `O(n)`.